DiscordBot/UnitTests/SeparateTextAndVoiceBasedChannels.json

```diff
@@ -0,0 +1,366 @@
+{
+  "name": "kevwjin's Server",
```

Comment:
you could create a JSON generation function and pass all of this data as parameters so all of this unit test data can be generated in less lines

Reply:
Oh, I should let you know that I've changed the JSON format a bit. I changed the key `text_channels` to `text_based_channels` and the key `voice_channels` to `voice_based_channels`. These are two classifications of channels. These two classifications of channels have types of channels. `text_based_channels` have the types `TEXT`, `ANCMT`, and `FORUM`, while `voice_based_channels` have the types `VOICE` and `STAGE`. These classifications are needed because in a category, `text_based_channels` are always ordered on top of `voice_based_channels`, e.g., `TEXT` channels are always ordered on top of `VOICE` and `STAGE` channels.